### DISCUSSION TITLE
Geolocation data in a Synchro based iModel

### DISCUSSION THREAD

**Discussion by oussamabonnor1** [iTwin/itwinjs-core#3196] [Q&A] [answered]
Is there a way to include geolocation data in an iModel that was generated from Synchro 4D Pro ?
This is desired to be able to display terrain tiles around the model.

**Comment by AlainRobertAtBentley** [comment, marked as answer]
@oussamabonnor1 The concepts related to geolocation are exposed here: https://www.itwinjs.org/learning/geolocation/
From your question I would assume that the current coordinates in your iModel are expressed relative to an arbitrary 3D coordinate system for which there does not exist a relation to Earth (sometimes refered as local grids). If this is the case then the 'Linear' geolocation is appropriate for you. This 'Linear' geolocation is specified in the iModel by setting the ecefLocation property (https://www.itwinjs.org/reference/core-common/imodels/imodel/eceflocation/). This property contains members that specify the origin of your coordinate system relative to the Earth in 3D with additional 3D orientation parameters (Yaw, Pitch, Roll) that can express the relation between your coordinate system and the Earth. 
Of course these properties are not particularly easy to compute but I believe there exists a tool that can compute these parameters based on the latitude and longitude of the origin of your 3D origin plus the direction of the North. I think @bbastings can help you about this tool.

**Reply by oussamabonnor1** [reply]
Thank you!
